python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+        params = bytearray(open(os.path.join(tmp_dir, "mod.params"), "rb").read())
+
+        if hostname:
+            # Remote RPC
+            if rpc_key:
+                logger.debug("running on remote RPC tracker with key %s", rpc_key)
+                remote = request_remote(rpc_key, hostname, port, timeout=1000)
+            else:
+                logger.debug("running on remote RPC with no key")
+                remote = rpc.connect(hostname, port)
+        else:
+            # Local
+            logger.debug("running a local session")
+            remote = rpc.LocalSession()
+
+        remote.upload(os.path.join(tmp_dir, "mod.so"))
+        remote_lib = remote.load_module("mod.so")
+
+        # TODO expand to other supported devices, as listed in tvm.rpc.client (@leandron)
+        logger.debug("remote device is %s", device)
+        ctx = remote.cpu() if device == "cpu" else remote.gpu()
+
+        if profile:
+            logger.debug("creating runtime with profiling enabled")
+            module = debug_runtime.create(graph, remote_lib, ctx, dump_root="./prof")
+        else:
+            logger.debug("creating runtime with profiling disabled")
+            module = runtime.create(graph, remote_lib, ctx)
+
+        logger.debug("load params into the runtime module")
+        module.load_params(params)
+
+        shape_dict, dtype_dict = get_input_info(graph, params)
+        inputs_dict = make_inputs_dict(inputs, shape_dict, dtype_dict, fill_mode)
+
+        logger.debug("setting inputs to the module")
+        module.set_input(**inputs_dict)
+
+        # Run must be called explicitly if profiling
+        if profile:
+            logger.debug("running the module with profiling enabled")
+            module.run()
+
+        # create the module time evaluator (returns a function)
+        timer = module.module.time_evaluator("run", ctx, 1, repeat=repeat)
+        # call the evaluator function to invoke the module and save execution times
+        prof_result = timer()
+        # collect a list of execution times from the profiling results
+        times = prof_result.results
+
+        logger.debug("collecting the output tensors")
+        num_outputs = module.get_num_outputs()
+        outputs = {}
+        for i in range(num_outputs):
+            output_name = "output_{}".format(i)
+            outputs[output_name] = module.get_output(i).asnumpy()
+
+        return outputs, times
+
+
+def get_top_results(outputs, max_results):
+    """Return the top n results from the output tensor.
+
+    This function is primarily for image classification and will
+    not necessarily generalise.
+
+    Parameters
+    ----------
+    outputs : dict
+        Outputs dictionary - {output_name: np.array}.
+    max_results : int
+        Number of results to return
+
+    Returns
+    -------
+    top_results : np.array
+        Results array of shape (2, n).
+        The first row is the indices and the second is the values.
```

Comment:
Why it is always "2"?

Reply:
That was being used mostly to models like MobileNet, ResNet, etc. that would give you a (2, n) output tensor. @mbaret can give more context.

Reply:
This feature is specifically for debugging image networks really, which happens to be very useful for us. The first row is the indices of the predicted objects (eg. from the imagenet classes) and the second row is the classification probability. That's why this is '2'. It allows you to very quickly check the sanity of an image classification network without having to dig into the output.npz.

Reply:
Thanks for the clarification. IIUC, it seems like we limit the applications to be CNN models and do not consider others like object detection or NLP models? If so please add the descriptions in the corresponding places.

Reply:
This particular feature only works for image classification networks as it was a useful shortcut for testing them. Maybe we could make that explicit in the help text?